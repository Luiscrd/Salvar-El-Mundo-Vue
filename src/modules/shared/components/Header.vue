<template>
  <nav class="navbar bg-primary toolbar-mov">
    <div class="container-fluid py-3">
      <img width="140" alt="Lc Logo" src="@/assets/logolcprograming.png" />
      <a href="https://github.com/Luiscrd" target="_blank" class="link-github">
        <span class="span-lc">LcrPrograming</span>
        <img src="@/assets/github_PNG40.png" alt="Github" class="logo-github">
      </a>
    </div>
  </nav>
  <div class="toolbar" role="banner">
    <img width="100" alt="Lc Logo" src="@/assets/logolcprograming3.png" />
    <span>Salvar al mundo - Vue</span>
    <img width="50" alt="Vue Logo" src="@/assets/logo_vue.png" />
    <div class="spacer"></div>
    <a href="https://github.com/Luiscrd" target="_blank" class="link-github">
      <span class="span-lc">LcrPrograming</span>
      <img src="@/assets/github_PNG40.png" alt="Github" class="logo-github">
    </a>
  </div>
</template>

<script>
export default {
  name: 'Header',
  props: {
  }
}
</script>

<style scoped>
p {
  margin: 0;
}

a,
a:visited,
a:hover {
  color: white;
  text-decoration: none;
}

a:hover {
  color: #341948;
}

.spacer {
  flex: 1;
}

.toolbar {
  position: relative;
  width: 100%;
  height: 90px;
  display: flex;
  align-items: center;
  background-color: #341948;
  color: white;
  font-weight: bold;
  font-size: 1.5em;
}

.toolbar-mov {
  display: none;
}

.toolbar img {
  margin: 0 16px;
}

.link-github {
  cursor: pointer;
  transition: 0.5s;
  display: flex;
  align-items: center;
}

.link-github:hover {
  opacity: 0.5;
}

.span-lc {
  position: relative;
  color: white;
}

.logo-github {
  width: 50px;
}

@media screen and (max-width: 420px) {
  .toolbar {
    display: none;
  }

  .toolbar-mov {
    display: flex;
    justify-content: space-around;
  }

  .logo-github {
    width: 30px;
    margin-left: 10px;
  }

  .link-github {
    font-weight: bold;
  }

}
</style>