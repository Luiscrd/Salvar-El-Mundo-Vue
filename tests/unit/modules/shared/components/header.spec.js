import { shallowMount } from '@vue/test-utils'
import Header from '@/modules/shared/components/Header.vue'

const mockStore = {
    getters: jest.fn()
}

jest.mock('vuex', () => ({
    useStore: () => mockStore
}))

describe('App Component', () => {

    let wrapper;

    beforeEach(() => {

        wrapper = shallowMount(Header)

    });

    test('Debe de hacer mach con el snapshot', () => {

        expect(wrapper.html()).toMatchSnapshot()

    })

})