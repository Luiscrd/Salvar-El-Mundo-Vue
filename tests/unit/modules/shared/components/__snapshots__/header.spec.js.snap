// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Debe de hacer mach con el snapshot 1`] = `
<div class="toolbar" role="banner"><img width="100" alt="Lc Logo" src=""><span>Salvar al mundo - Vue</span><img width="50" alt="Vue Logo" src="">
  <div class="spacer"></div><a href="https://github.com/Luiscrd" target="_blank" class="link-github"><span class="span-lc">LcrPrograming</span><img src="" alt="Github" class="logo-github"></a>
</div>
`;
